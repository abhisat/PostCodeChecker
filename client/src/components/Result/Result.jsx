import React from 'react';
import ReactDOM from 'react-dom';
import styles from './style.module.css';
import { Typography } from '@material-ui/core';

const Result = (props) => {
    const { error, resultText } = props;
    
    return (
        <div className={resultText ? styles.message : styles.noshow}>
            <Typography className={styles.header}>
                Results
            </Typography>
            <Typography className={error ? styles.error: styles.result}>
                {resultText}
            </Typography>
        </div>
    );
};

export { Result };