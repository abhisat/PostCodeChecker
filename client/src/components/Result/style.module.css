.message {
    width: 80%;
    height: 10%;
    margin: 0 auto !important;
    margin-top: 10px !important;
    text-align: center;
    display: block;
}

.noshow {
    display: none;
}

.header{
    font-size: 1.5rem !important;
}

.error {
    color: red;
    background-color: rgb(241, 198, 198, 0.4);
    border: 0.5px solid red;
}

.result{
    color: #0075bf;
    background-color: #c6ddeb;
    border: 0.5px solid #0075bf;
}