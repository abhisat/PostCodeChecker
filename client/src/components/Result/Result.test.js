import React from "react";
import { Result } from "./Result";
import Enzyme, { mount, unmount } from 'enzyme';
import toJson from 'enzyme-to-json';
import EnzymeAdapter from 'enzyme-adapter-react-16';

Enzyme.configure({
  adapter: new EnzymeAdapter(),
});

const errorProps = {
  error: true,
  resultText: 'This is an error test.'
}

const resultProps = {
  error: false,
  resultText: 'This is a result test.'
}

const mountResult = (props = {}) => {
    const wrapper = mount(<Result {...props} />);
    return wrapper;
};

describe("<Results />", () => {
  it("Matches the snapshot", () => {
    const Result = mountResult();
    expect(toJson(Result)).toMatchSnapshot();
  });

  it('displays errors in red', () => {
    const Result = mountResult(errorProps);
    expect(Result.find('p.error')).toHaveLength(1);
    expect(Result.find('p.error').text()).toEqual('This is an error test.');
  });

  it('displays results in blue', () => {
    const Result = mountResult(resultProps);
    expect(Result.find('p.result')).toHaveLength(1);
    expect(Result.find('p.result').text()).toEqual('This is a result test.');
  })
});