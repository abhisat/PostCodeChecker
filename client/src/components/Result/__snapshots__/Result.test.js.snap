// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> Matches the snapshot 1`] = `
<div
  className="noshow"
>
  <WithStyles(ForwardRef(Typography))
    className="header"
  >
    Results
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    className="result"
  />
</div>
`;
